### PR TITLE
Edit reminder and save 

### LINKED ISSUE #9: Display Default Page
Update reminder component to have a conditional that will only display notes if notes exist.

There should be a default welcome page welcoming users to the app and/or directions to create a new note. Feel free to be creative here.

### LINKED ISSUE #10: Editing A Note
When viewing an existing reminder, the user can click on an “Edit” button where they can modify the title, notes, or date of the reminder.

* When editing a note, there is a “Save” button that commits the changes to the server.
* When the user hits the save button, they return to viewing that individual note.

Things to keep in mind:
* What should your route look like to edit a reminder?
* Next Steps: What are similarities between the new and edit routes that we can refactor?

### PR DESCRIPTION
**Pull request by gness1804**
## Purpose

_Describe the problem or feature in addition to a link to the issues. Delete any content that doesn't apply._

## Approach

The user can now edit an existing reminder, which then renders to the page in its modified form. There is an "edit" button on each reminder that allows such editing, and a "submit" button in the edit view confirms the changes and sends the user back to the non-edit view. Also adds the `bootstrap-datepicker` plugin to allow dates to update seamlessly. 

Closes #9 
Closes #10 

_How does this change address the problem?_

Reminders are now editable, so if the user made a mistake or circumstances change, they can update the reminder. 

### Learning

_Describe the research stage._

_Links to blog posts, patterns, libraries or add-ons used to solve this problem._

We consulted Ember documentation on issues such as controllers v. components, how to edit items, and the proper use of links and actions. We also consulted peers to help us through  tough spots. 

https://github.com/soulim/ember-cli-bootstrap-datepicker
https://guides.emberjs.com/v1.10.0/controllers/representing-a-single-model-with-objectcontroller/
https://guides.emberjs.com/v1.10.0/getting-started/toggle-todo-editing-state/
https://guides.emberjs.com/v2.5.0/templates/links/
https://guides.emberjs.com/v2.5.0/components/triggering-changes-with-actions/

_Briefly describe what tests you have written and why._

Two acceptance tests check whether a user can enter in information via an edit and whether the edit forms render when the user clicks the edit button. One of the tests also ensures that the original information renders when user enters it in for a new reminder.

### CODE REVIEW

File: app/controllers/reminders/edit.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  actions: {
+    editReminder(title, date, notes, id) {
+      console.log('foo');
```
Comment (martensonbj) [inline app/controllers/reminders/edit.js]: Please remove console logs before submitting a PR

File: app/templates/reminders/edit.hbs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<form class="edit-reminder-form">
+  {{input class="edit-reminder-title" value=model.title}}
+  {{bootstrap-datepicker class="edit-reminder-date" value=model.date}}
+  {{input class="edit-reminder-notes" value=model.notes}}
+  {{#link-to 'reminders.reminder' model}}
+    <button class="submit-edits-button" {{action 'editReminder' model.title model.date model.notes model.id on="submit"}}>Save</button>
```
Comment (martensonbj) [inline app/templates/reminders/edit.hbs]: Simply pass in the `model` here and not each individual `model.property` - those should be pulled out in the controller.

File: tests/unit/routes/edit-test.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { moduleFor, test } from 'ember-qunit';
```
Comment (martensonbj) [inline tests/unit/routes/edit-test.js]: We don't need this file

**Comment by gness1804** [comment]
Here are the requested changes.

**Pull request merged**